app/controllers/review_offer.js

```diff
@@ -13,6 +13,44 @@ export default Ember.Controller.extend(AsyncTasksMixin, {
   displayCompleteReviewPopup: false,
   displayCompleteReceivePopup: false,
 
+  allShareables: Ember.computed(function() {
+    return this.get("store").peekAll("shareable");
+  }),
+
+  allMessages: Ember.computed(function() {
+    return this.get("store").peekAll("message");
+  }),
+
+  offerShareable: Ember.computed(
+    "offer.id",
+    "allShareables.[]",
+    "allShareables.@each.active",
+    function() {
+      return this.get("allShareables").find(sh => {
+        return (
+          sh.get("resourceType") == "Offer" &&
```

Comment:
"Offer", this string check is made many times, probably u can add that in a constant. Not mandatory though

Reply:
oh yeah, I'm rushing through